gtda/mapper/tests/test_visualization.py

```diff
@@ -89,14 +91,93 @@ def test_unsuitable_colorscale_for_hoverlabel_3d(X):
+    pipe = make_mapper_pipeline()
+
+    with pytest.raises(ValueError):
+        plot_static_mapper_graph(pipe, X_arr, color_data=X_arr[:-1])
+
+
+@pytest.mark.parametrize("color_features",
+                         [PCA(n_components=2),
+                          PCA(n_components=2).fit(X_arr).transform])
+def test_color_features_as_estimator_or_callable(color_features):
     pipe = make_mapper_pipeline()
     graph = pipe.fit_transform(X_arr)
-    node_color_statistic = np.arange(len(graph.vs))
+    node_elements = graph.vs["node_elements"]
+
+    pca = PCA(n_components=2)
+    color_data_transformed = pca.fit_transform(X_arr)
+    node_colors_color_features = \
+        np.array([np.mean(color_data_transformed[itr, 0])
+                  for itr in node_elements])
+
+    fig = plot_static_mapper_graph(pipe, X_arr, color_data=X_arr,
+                                   color_features=color_features)
+
+    assert_almost_equal(fig.data[1].marker.color, node_colors_color_features)
+
+
+@pytest.mark.parametrize("color_features", [X_arr, X_df])
+def test_invalid_color_features_types(color_features):
+    pipe = make_mapper_pipeline()
+
+    with pytest.raises(ValueError):
+        plot_static_mapper_graph(pipe, X_arr,
+                                 color_features=color_features)
+
+
+@pytest.mark.parametrize("is_2d", [False, True])
+def test_node_color_statistic_as_ndarray(is_2d):
+    pipe = make_mapper_pipeline()
+    graph = pipe.fit_transform(X_arr)
+    node_color_statistic_col_0 = np.arange(len(graph.vs))
+    if is_2d:
+        node_color_statistic = np.vstack([node_color_statistic_col_0,
+                                          node_color_statistic_col_0]).T
+    else:
+        node_color_statistic = node_color_statistic_col_0
+
     fig = plot_static_mapper_graph(pipe, X_arr,
                                    node_color_statistic=node_color_statistic)
 
-    assert np.array_equal(fig.data[1].marker.color, node_color_statistic)
+    assert np.array_equal(fig.data[1].marker.color, node_color_statistic_col_0)
+
+
+def test_node_color_statistic_as_ndarray_wrong_length():
+    pipe = make_mapper_pipeline()
+    graph = pipe.fit_transform(X_arr)
+    node_color_statistic = np.arange(len(graph.vs) + 1)
+
+    with pytest.raises(ValueError):
+        plot_static_mapper_graph(pipe, X_arr,
+                                 node_color_statistic=node_color_statistic)
+
+
+def test_invalid_type_node_color_statistic_static():
+    pipe = make_mapper_pipeline()
+
+    with pytest.raises(ValueError):
+        plot_static_mapper_graph(pipe, X_arr, node_color_statistic="foo")
+
+
+def test_invalid_node_color_statistic_interactive():
+    pipe = make_mapper_pipeline()
+    graph = pipe.fit_transform(X_arr)
+    node_color_statistic = np.arange(len(graph.vs))
+    with pytest.raises(ValueError):
+        plot_interactive_mapper_graph(
+            pipe, X_arr, node_color_statistic=node_color_statistic
+            )
```

Comment:
Why is this `node_color_statistic` invalid? According to the docs, one should be able to pass an `ndarray` of shape `(n_nodes,)`

Reply:
Thinking about it, since the graph is susceptible to change, should we really allow passing ndarray as `node_color_statistic` in the interactive plotting?

Reply:
We don't allow it. That's what I am testing here.

Reply:
> According to the docs, one should be able to pass an ndarray of shape (n_nodes,)
Not according to the docs for interactive plotting :)

Reply:
Indeed - i must have gotten lost. Solved!